### PR TITLE
ARROW-11849: [R] Use roxygen @examplesIf

### PR DESCRIPTION
**Pull request by thisisnic**
Swap out dontrun and donttest tags in examples and replace with examplesIf

### CODE REVIEW

**Comment by thisisnic** [comment]
@github-actions crossbow submit test-r-minimal-build

**Comment by ianmcook** [comment]
@thisisnic could you please add a note in `r/vignettes/developing.Rmd` instructing folks who want to contribute to the package to install roxygen2 from GitHub using `remotes::install_github("r-lib/roxygen2")`? I think this should go someplace under the **Developer environment setup** heading.

File: r/R/filesystem.R [hunk 1]
```diff
@@ -476,15 +476,13 @@ SubTreeFileSystem$create <- function(base_path, base_fs = NULL) {
 #' copying but may help accommodate high latency FileSystems.
 #' @return Nothing: called for side effects in the file system
 #' @export
-#' @examples
-#' \dontrun{
+#' @examplesIf FALSE
```
Comment (ianmcook) [inline r/R/filesystem.R:479]: I think it's best to keep `\dontrun{}` in cases where the conditional is a literal `FALSE`

Comment (ianmcook) [reply r/R/filesystem.R:479]: On second thought, disregard this—I hadn't realized that you eliminated _all_ the other instances of `\dontrun{}` so for consistency it makes sense to do this one too.

**Comment by ianmcook** [comment]
If we're doing this, let's go ahead and do it for `\donttest{}` too. I just pushed a commit doing that (8df7c51).

**Comment by ianmcook** [comment]
@github-actions crossbow submit test-r-minimal-build

**Comment by ianmcook** [comment]
I ran `R CMD check --as-cran` locally (a) with no C++ library available, and (b) with a very minimally built C++ library and all the checks passed

**Comment by ianmcook** [comment]
Thanks @thisisnic, this looks good to me!

There was some discussion about whether we should include this in `DESCRIPTION`:

```
Suggests: 
    roxygen2
Remotes:  
    r-lib/roxygen2
```
Were we to do that, the existing instruction to run `remotes::install_deps(dependencies = TRUE)` (in `developing.Rmd`) would install `roxygen2` from GitHub. 

Any opinions on the pros and cons of doing that vs. simply keeping the instruction to run `remotes::install_github("r-lib/roxygen2")` that is added to `developing.Rmd` in this PR?

cc @jonkeane

**Comment by ianmcook** [comment]
Re whether or not we should modify `DESCRIPTION`: After discussing that with @thisisnic, I think we can merge this as is and consider that question separately in the context of ARROW-12743

**Review by ianmcook — APPROVED** [review]

**Comment by jonkeane** [comment]
Yeah, I think this is fine to merge. One thing to note, that version number requirement might be satisfied but with a dev version that doesn't include the feature we're using (I haven't checked the git history, but I imagine there are commits after the last release, but before they added `examplesif`). Which would be annoying to diagnose, but with our comment about needing an up-to-date roxygen2, that should hopefully be enough.  I'll comment on the jira with thoughts about `DESCRIPTION` itself.

Do we already have or do we need/want a CI job that checks if the docs can be built when these features aren't around — I assume that `examplesif` works just fine, but it would be good for us to check that we've not accidentally forgotten it / used the wrong test / etc. (this can of course also be a follow on)

**Comment by ianmcook** [comment]
> Do we already have or do we need/want a CI job that checks if the docs can be built when these features aren't around — I assume that `examplesif` works just fine, but it would be good for us to check that we've not accidentally forgotten it / used the wrong test / etc. (this can of course also be a follow on)

Perhaps this could be incorporated into a CI task that also checks that the `Rd` files have been built. I believe we don't currently have that, and several times recently PRs have been merged without the docs being built. Maybe we could achieve this by adding to an existing CI task instead of adding a new one.

**Comment by jonkeane** [comment]
Yes, adding to an existing CI job would be great. Though the thing I was wondering about testing is running examples with a minimal build to make sure they skip appropriately. I think we could add that on to the minimal build CI job (and then maybe also add on a run-the-examples step to one of the more maximal builds to make sure they run when things *are* available?). Checking for missing Rds would be great (and maybe could be part of the script that checks for missing topics from [ARROW-12236](https://github.com/ursacomputing/crossbow/actions/runs/879127390)?)

**Pull request closed without merging**